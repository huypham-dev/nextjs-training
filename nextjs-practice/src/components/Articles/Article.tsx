import { memo } from "react";

import Link from "next/link";
import Image from "next/image";
import { PostType } from "types";
import isEqual from "react-fast-compare";

type Props = {
  article: PostType;
};

const Article = ({ article }: Props) => {
  const { id, image, title, category, content } = article;

  return (
    <article className="max-w-md mx-auto rounded-md shadow-md overflow-hidden md:max-w-2xl mb-12">
      <div className="md:flex">
        <Link href={`/post/${id}`}>
          <a className="flex flex-shrink-0 justify-center md:items-center">
            <Image src={image} alt="article image" width={250} height={150} />
          </a>
        </Link>
        <div className="p-8">
          <div className="uppercase tracking-wide text-sm text-indigo-500 font-semibold">
            {category}
          </div>
          <Link href={`/post/${id}`}>
            <a className="block mt-1 text-lg leading-tight font-medium text-black hover:underline">
              {title}
            </a>
          </Link>
          <p className="mt-2 text-gray-500">{content.substring(0, 100)}...</p>
        </div>
      </div>
    </article>
  );
};

export default memo(Article, isEqual);
