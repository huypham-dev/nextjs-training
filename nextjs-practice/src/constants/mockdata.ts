export const CATEGORIES = [
  {
    id: "1",
    name: "mobile",
    image:
      "https://images.prismic.io/nextjsexample/7853f863-6a12-4312-a6da-f930437f6d62_mobile.png",
  },
  {
    id: "2",
    name: "internet",
    image:
      "https://images.prismic.io/nextjsexample/cb71ce70-bb03-442b-99e8-a318255b81f3_3579187.png",
  },
];

export const ARTICLES = [
  {
    id: "1",
    createdAt: "2021-03-01T12:59:16.855Z",
    title: "Lorem ipsum dolor sit amet, consectetur adipisicing elit",
    author: "author 1",
    image:
      "https://genk.mediacdn.vn/139269124445442048/2021/3/2/photo-1-16146533712881100681556.jpg",
    category: "internet",
    content:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.1",
    slug: "slug 1",
    view: 90,
  },
  {
    id: "2",
    createdAt: "2021-03-01T12:59:16.855Z",
    title: "Lorem ipsum dolor sit amet, consectetur adipisicing elit",
    author: "author 1",
    image:
      "https://genk.mediacdn.vn/139269124445442048/2021/3/2/photo-1-16146533712881100681556.jpg",
    category: "internet",
    content:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.1",
    slug: "slug 1",
    view: 90,
  },
];
