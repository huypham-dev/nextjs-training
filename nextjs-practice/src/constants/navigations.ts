export const navigations = [
  { label: "Home", path: "/" },
  { label: "Blog", path: "/blog" },
  { label: "Contact", path: "/contact" },
];
