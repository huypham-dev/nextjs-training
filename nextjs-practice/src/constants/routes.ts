export const ROUTES = {
  home: {
    path: "/",
  },
  blog: {
    path: "/blog",
    withId: (id: string) => `/blog/${id}`,
  },
  post: {
    path: "/post",
    withId: (id: string) => `/post/${id}`,
  },
  contact: {
    path: "/",
  },
};
