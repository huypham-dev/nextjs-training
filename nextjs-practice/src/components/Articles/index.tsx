import { memo } from "react";
import { PostType } from "types";
import Article from "./Article";

type Props = {
  data: PostType[];
};

const Articles = ({ data = [] }: Props) => (
  <div className="w-full md:pr-12 mb-12">
    {data.map((item) => (
      <Article key={item.id} article={item} />
    ))}
  </div>
);

export default memo(Articles);
