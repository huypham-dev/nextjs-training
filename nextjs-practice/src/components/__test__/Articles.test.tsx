import { mount, shallow } from "enzyme";
import Articles from "../Articles/index";
import { mockArticles } from "constants/mockdata";

describe("Articles", () => {
  it("Snapshot", () => {
    const component = shallow(<Articles data={[]} />);

    expect(component).toMatchSnapshot();
  });

  it("Render component correctly with props", () => {
    const component = mount(<Articles data={mockArticles} />);
    const articleList = component.find("article");

    expect(articleList.length).toBe(2);
  });
});
