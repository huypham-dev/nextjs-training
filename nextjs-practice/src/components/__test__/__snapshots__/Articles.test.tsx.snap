// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles Snapshot 1`] = `
<div
  className="w-full md:pr-12 mb-12"
/>
`;
